### ISSUE TITLE
`ActiveRecord::Encryption` concerns; `:body` column from `EncryptedRichText` appears twice in `filter_parameters` and causes generic parameter names to be filtered;

### ISSUE THREAD

**Issue by nvasilevski** [rails/rails#44330]
### Steps to reproduce

Sorry I'm not able to provide a reproduction step because most likely loading the full app is necessary in order to reproduce the issue:

1. `rails new`
2. load the app or `rails c` with `eager_load = true` 
3.  `ActiveRecord::Encryption.encrypted_attribute_declaration_listeners` contains two the same hooks
And because we have two same hooks declared, we do append the `:body` parameter name twice:
```ruby
3.0.3 :004 > Rails.application.config.filter_parameters
 => [:password, :body, :body]
```

The `:body` param is coming from here:
https://github.com/rails/rails/blob/638a92f7344963c7e42eeb84eb816e4592bda7e4/actiontext/app/models/action_text/encrypted_rich_text.rb#L7

### Secondary concern

Another issue I wanted to discuss is that having such a generic parameter name included almost "by default" causes many unrelated parameters to be filtered. Should we consider using the "dot notation" in order to scope filtering per `encrypted_rich_text.body` ?
https://github.com/rails/rails/blob/167f5c80658a15b5aa2524433affe52553e2c8d9/activesupport/lib/active_support/parameter_filter.rb#L8-L9

For example:

```ruby
pf = ActiveSupport::ParameterFilter.new([:body])

my_hash_with_various_unrelated_params = {
  response_body: "response",
  request_body: "request",
  is_my_body_cool: true,
  body: "i'm an encrypted rich text body",
  nested_key: { body_here: "hey", somebody_here: true }
};

pf.filter(my_hash_with_various_unrelated_params)

 =>
{:response_body=>"[FILTERED]",
 :request_body=>"[FILTERED]",
 :is_my_body_cool=>"[FILTERED]",
 :body=>"[FILTERED]",
 :nested_key=>{:body_here=>"[FILTERED]", :somebody_here=>"[FILTERED]"}}
```

Let me know if it would be better to open a separate issue to discuss the filtered params behaviour. Thanks!

### Expected behavior

`ActiveRecord::Encryption.encrypted_attribute_declaration_listeners` should have only one listener by default, or at least the one defined at:
https://github.com/rails/rails/blob/18707ab17fa492eb25ad2e8f9818a320dc20b823/activerecord/lib/active_record/encryption/configurable.rb#L54-L56

Should present only once

### Actual behavior

We do have the same hook declared twice:
```ruby
3.0.3 :001 > ActiveRecord::Encryption.encrypted_attribute_declaration_listeners
 =>
[#<Proc:0x000056129801aa30 /usr/share/rvm/gems/ruby-3.0.3/gems/activerecord-7.0.1/lib/active_record/encryption/configurable.rb:54>,
 #<Proc:0x0000561297fbacc0 /usr/share/rvm/gems/ruby-3.0.3/gems/activerecord-7.0.1/lib/active_record/encryption/configurable.rb:54>]
```
### System configuration
**Rails version**: `"7.0.1"` but most likely reproducible with the introduction of encrypted attributes

**Ruby version**: `3.0.3`

**Comment by nvasilevski** [comment]
Update:
Removing `jbuilder` from the gemfile fixes the duplicated hook issue.
If I'm not mistaken, the second load is coming from here:
https://github.com/rails/jbuilder/blob/main/lib/jbuilder/railtie.rb#L20

**Comment by jorgemanrubia** [comment]
Yup @rafaelfranca. I'll look into it this week.

**Comment by nvasilevski** [comment]
Thanks! 
I think I have a fix for the duplicated `:body` value - https://github.com/rails/rails/pull/44355

**Comment by nvasilevski** [comment]
Using the chance just wanted to give a little bit more visibility to this PR - https://github.com/rails/rails/pull/44329/files
It's not directly related to the problem but fixes a tiny issue very close to the area we are discussing in this issue
